src/Microsoft.ML.Data/Scorers/FeatureContributionCalculationTransform.cs

```diff
@@ -0,0 +1,497 @@
+    /// ]]>
+    /// </format>
+    /// </example>
+    public sealed class FeatureContributionCalculationTransform
+    {
+        // Apparently, loader signature is limited in length to 24 characters.
+        internal const string MapperLoaderSignature = "WTFBindable";
+        private const int MaxTopBottom = 1000;
+
+        public sealed class Arguments : ScorerArgumentsBase
+        {
+            [Argument(ArgumentType.AtMostOnce, HelpText = "Number of top contributions", SortOrder = 1)]
+            public int Top = 10;
+
+            [Argument(ArgumentType.AtMostOnce, HelpText = "Number of bottom contributions", SortOrder = 2)]
+            public int Bottom = 10;
+
+            [Argument(ArgumentType.AtMostOnce, HelpText = "Whether or not output of Features contribution should be normalized", ShortName = "norm", SortOrder = 3)]
+            public bool Normalize = true;
+
+            [Argument(ArgumentType.AtMostOnce, HelpText = "Whether or not output of Features contribution in string key-value format", ShortName = "str", SortOrder = 4)]
+            public bool Stringify = false;
+
+            // REVIEW: the scorer currently ignores the 'suffix' argument from the base class. It should respect it.
+        }
+
+        public static IDataScorerTransform Create(IHostEnvironment env, Arguments args, IDataView data, ISchemaBoundMapper mapper, RoleMappedSchema trainSchema)
+        {
+            Contracts.CheckValue(env, nameof(env));
+            env.CheckValue(data, nameof(data));
+            env.CheckValue(mapper, nameof(mapper));
+            if (args.Top <= 0 || args.Top > MaxTopBottom)
+                throw env.Except($"Number of top contribution must be in range (0,{MaxTopBottom}]");
+            if (args.Bottom <= 0 || args.Bottom > MaxTopBottom)
+                throw env.Except($"Number of bottom contribution must be in range (0,{MaxTopBottom}]");
+
+            var contributionMapper = mapper as RowMapper;
+            env.CheckParam(mapper != null, nameof(mapper), "Unexpected mapper");
+
+            var scorer = ScoreUtils.GetScorerComponent(env, contributionMapper);
+            var scoredPipe = scorer.CreateComponent(env, data, contributionMapper, trainSchema);
+            return scoredPipe;
+        }
+
+        public static ISchemaBindableMapper Create(IHostEnvironment env, Arguments args, IPredictor predictor)
+        {
+            Contracts.CheckValue(env, nameof(env));
+            env.CheckValue(args, nameof(args));
+            env.CheckValue(predictor, nameof(predictor));
+            if (args.Top <= 0 || args.Top > MaxTopBottom)
+                throw env.Except($"Number of top contribution must be in range (0,{MaxTopBottom}]");
+            if (args.Bottom <= 0 || args.Bottom > MaxTopBottom)
+                throw env.Except($"Number of bottom contribution must be in range (0,{MaxTopBottom}]");
```

Comment:
minor nit, I would move this into a helper function. ValidateTopBottomArguments?